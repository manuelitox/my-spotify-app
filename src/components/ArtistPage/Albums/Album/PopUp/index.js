import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import Track from 'components/generic/Track/'
import PopUp from 'components/generic/PopUp/container'
import PlaceholderAlbum from 'svgs/placeholderAlbum.svg'
import StylesHeaderAlbum from './styles-header'
import StylesWrapperTracks from './styles-wrapper-tracks'

const PopUpAlbum = ({ isOpen, togglePopUp, cover, title, releaseDate, numberTracks }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <StylesHeaderAlbum>
        <img 
          src={ cover ? cover : PlaceholderAlbum } 
          alt={ title }
          width="300"
          height="300" />
        <h3>{ title }</h3>
        <p>Total tracks: { numberTracks }</p>
        <p>Release date: { releaseDate }</p>
      </StylesHeaderAlbum>
      <StylesWrapperTracks>
        <Track 
          name="Empire State of Mind" 
          duration="3 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="No one" 
          duration="4 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />                
      </StylesWrapperTracks>
    </Fragment>
  </PopUp>      
)

PopUpAlbum.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  cover: PropTypes.string,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  numberTracks: PropTypes.number.isRequired
}

export default PopUpAlbum