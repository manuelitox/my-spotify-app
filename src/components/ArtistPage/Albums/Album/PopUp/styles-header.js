import styled from 'styled-components'

const StylesHeaderAlbum = styled.div`
  margin: 30px 0;
  img {
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      height: auto;
      max-width: 300px;
      width: 100%;
    }
  }
  h3 {
    color: ${ props => props.theme.blueDark };
    font-size: 1.6rem;
    font-weight: 500;
    margin: 10px 0 4px;
  }
  p {
    color: ${ props => props.theme.offWhite };
    font-size: 1.4rem;
    line-height: 1.6;
    margin: 0;
  }
`

export default StylesHeaderAlbum