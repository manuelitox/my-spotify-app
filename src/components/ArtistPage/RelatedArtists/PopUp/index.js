import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Artist from 'components/ArtistPage/RelatedArtists/Artist/'
import StyledWrapperRelatedArtists from 'components/ArtistPage/RelatedArtists/styles-wrapper'
import getImage from 'lib/get-image'

const PopUpRelatedArtists = ({ isOpen, togglePopUp, relatedArtists }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Related Artists</Title>
      <StyledWrapperRelatedArtists inPopup onClick={ togglePopUp }>
        { relatedArtists.map((artist, index) =>Â (
          <Artist key={ index } id={ artist.id } photo={ getImage(artist.images, 200) } name={ artist.name } />
        )) }   
      </StyledWrapperRelatedArtists>
    </Fragment>
  </PopUp>      
)

PopUpRelatedArtists.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  relatedArtists: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      images: PropTypes.array      
    })
  ).isRequired
}

export default PopUpRelatedArtists