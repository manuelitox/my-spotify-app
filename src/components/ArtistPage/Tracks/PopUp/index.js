import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import Title from 'components/ArtistPage/Title/'
import Track from 'components/generic/Track/'
import PopUp from 'components/generic/PopUp/container'
import StylesWrapperTracks from 'components/ArtistPage/stylesWrapperTracks'

const PopUpTopTracks = ({ isOpen, togglePopUp }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Top Tracks</Title>
      <StylesWrapperTracks>
        <Track 
          name="Empire State of Mind" 
          duration="3 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="No one" 
          duration="4 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />                
      </StylesWrapperTracks>
    </Fragment>
  </PopUp>      
)

PopUpTopTracks.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired
}

export default PopUpTopTracks