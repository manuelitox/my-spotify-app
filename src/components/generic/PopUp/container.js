import React, { Component } from 'react'
import PropTypes from 'prop-types'
import onClickOutside from 'react-onclickoutside'

import PopUp from './'

export class PopUpContainer extends Component {
  handleClickOutside = () => this.props.togglePopUp()

  render () {
    return this.props.isOpen && <PopUp { ...this.props } />
  }
}

PopUpContainer.defaultProps = {
  WithCloseButton: true
}

PopUpContainer.propTypes = {
  width: PropTypes.string, 
  isOpen: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.object
  ]),
  togglePopUp: PropTypes.func.isRequired,
  WithCloseButton: PropTypes.bool
}

export default onClickOutside(PopUpContainer)