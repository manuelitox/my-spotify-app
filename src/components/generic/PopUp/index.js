import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import BodyClassName from 'react-body-classname'

import theme from 'config/theme'
import Button from 'components/generic/Button/'
import StyledPopUp from './styles'

const PopUp = ({ width, children, togglePopUp, textCloseButton, WithCloseButton }) => (
  <Fragment>
    <BodyClassName className="fixed" />
    <StyledPopUp width={ width }>
      { WithCloseButton ? 
        <Button 
          bg={ theme.blueDark }
          textColor="white"
          onClick={ togglePopUp }>
          { textCloseButton ? textCloseButton : 'Close' }
        </Button> : undefined }
      { children }
    </StyledPopUp>
  </Fragment>
)

PopUp.propTypes = {
  width: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  togglePopUp: PropTypes.func.isRequired,
  WithCloseButton: PropTypes.bool
}

export default PopUp