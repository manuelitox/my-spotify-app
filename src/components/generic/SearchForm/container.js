import React, { Component } from 'react'
import onClickOutside from 'react-onclickoutside'

import SearchForm from './base'

export class SearchFormContainer extends Component {
  state = {
    isOpen: false,
    loading: false,
    suggestions: []
  }

  handleClickOutside = evt => {
    this.setState({ isOpen: false })
  }

  openSuggestions = () => {
    const { suggestions } = this.state
    if ( suggestions.length >= 1 ) this.setState({ isOpen: true })
  }

  onChange = e => {
    console.log(e.target.value)
    this.setState({ 
      isOpen: true,
      loading: true 
    })
    setTimeout(() => {
      this.setState({ 
        suggestions: [
          { id: '1', name: 'Alicia Keys' },
          { id: '2', name: 'Queen' },
          { id: '3', name: 'John Mayer' },
          { id: '4', name: 'ACDC' },
          { id: '5', name: 'Blink 182' },
          { id: '6', name: 'Bon Jovi' },
          { id: '7', name: '2pac' }          
        ],
        loading: false
    })
    }, 1000);
  }

  render () {
    return (
      <SearchForm 
        onChange={ this.onChange } 
        openSuggestions={ this.openSuggestions } 
        { ...this.state } />
    )
  }
}

export default onClickOutside(SearchFormContainer)