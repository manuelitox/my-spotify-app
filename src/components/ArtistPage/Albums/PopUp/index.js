import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Album from 'components/ArtistPage/Albums/Album/'
import StyledWrapperAlbums from 'components/ArtistPage/Albums/styles'
import getImage from 'lib/get-image'

const PopUpAlbums = ({ isOpen, togglePopUp, albums, className }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp } className={ className }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Albums</Title>
      <StyledWrapperAlbums inPopup>
        { albums.map( (album, index) => (
          <Album key={ index } id={ album.id } title={ album.name } cover={ getImage(album.images) } numberTracks={ album.total_tracks } releaseDate={ album.release_date } />
        )) }
      </StyledWrapperAlbums>
    </Fragment>
  </PopUp>      
)

PopUpAlbums.propTypes = {
  isOpen: PropTypes.bool,
  className: PropTypes.string,
  togglePopUp: PropTypes.func.isRequired,
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      total_tracks: PropTypes.number,
      release_date: PropTypes.string
    })
  )
}

export default PopUpAlbums