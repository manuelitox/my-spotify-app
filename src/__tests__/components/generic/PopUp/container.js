import React from 'react'
import { shallow } from 'enzyme'
import BodyClassName from 'react-body-classname'

import { PopUpContainer } from 'components/generic/PopUp/container'

describe('Component | PopUpContainer', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <PopUpContainer togglePopUp={ jest.fn() }>
        <p>PopUp!!</p>
      </PopUpContainer>
    )
  })
  it('should not render because isOpen is false by default', () => {
    expect(component.contains(<BodyClassName className="fixed" />)).toBeFalsy()
    expect(component.find('p').length).toEqual(0)
    expect(component).toMatchSnapshot()
  })
  it('should render a PopUpContainer', () => {
    component = shallow(<PopUpContainer togglePopUp={ jest.fn() } isOpen><p>PopUp!!</p></PopUpContainer>)
    expect(component).toMatchSnapshot()
  })
})