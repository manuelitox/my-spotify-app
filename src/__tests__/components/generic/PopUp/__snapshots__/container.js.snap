// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | PopUpContainer should not render because isOpen is false by default 1`] = `""`;

exports[`Component | PopUpContainer should render a PopUpContainer 1`] = `
<PopUp
  WithCloseButton={true}
  isOpen={true}
  togglePopUp={[MockFunction]}
>
  <p>
    PopUp!!
  </p>
</PopUp>
`;
